open_prices/proofs/ml.py

```diff
@@ -843,14 +869,43 @@ def run_and_save_price_tag_extraction(
     else:
         responses = [extract_from_price_tag(image) for image in preprocessed_images]
     for price_tag, response in zip(price_tags, responses):
+        if response.parsed is None:
+            logger.info(
+                "Failed to extract price tag for price tag id %s: %s",
+                price_tag.id,
+                response.text,
+            )
+            continue
+
+        barcode = response.parsed.barcode
+        similar_barcodes = []
+        # barcode similarity search is quite costly (500~1000ms for 4M
+        # products), so we only run it if the barcode doesn't exist in the
+        # database
+        if barcode and not Product.objects.filter(code=barcode).exists():
+            similar_barcodes = [
```

Comment:
shouldn't we return only similar barcodes that are "valid" ?
running them in `common_openfoodfacts.barcode_is_valid()`